core/modules/book/book.module

```diff
@@ -87,11 +87,13 @@ function book_autoload_info() {
  */
 function book_node_view_link(Node $node, $view_mode) {
   $links = array();
+  $config = config('book.settings');
+  $link_options = $config->get('book_links');
 
   if (isset($node->book['depth'])) {
     if ($view_mode == 'full' && node_is_page($node)) {
-      $child_type = config_get('book.settings', 'book_child_type');
-      if ((user_access('add content to books') || user_access('administer book outlines')) && node_access('create', $child_type) && $node->status == 1 && $node->book['depth'] < MENU_MAX_DEPTH) {
+      $child_type = $config->get('book_child_type');
+      if ((user_access('add content to books') || user_access('administer book outlines')) && node_access('create', $child_type) && $node->status == 1 && $node->book['depth'] < MENU_MAX_DEPTH && in_array('book_add_child', $link_options, TRUE)) {
```

Comment:
The logic in this if statement is getting hard to follow. Instead of adding more conditions to an already long line, can we rewrite this as something like...

```php
if (user_access('add content to books') || user_access('administer book outlines')) {
  $link_options = $config->get('book_links');
  $create_access = node_access('create', $child_type);
  $depth_allowed =  ($node->book['depth'] < MENU_MAX_DEPTH);
  $show_add_child_link = in_array('book_add_child', $link_options, TRUE);
  if ( $create_access && $node->status == 1 &&  $depth_allowed && $show_add_child_link) {
    ...
  }
}
```